frame/treasury/src/lib.rs

```diff
@@ -404,7 +408,9 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 
 	/// The needed bond for a proposal whose spend is `value`.
 	fn calculate_bond(value: BalanceOf<T, I>) -> BalanceOf<T, I> {
-		T::ProposalBondMinimum::get().max(T::ProposalBond::get() * value)
+		let mut r = T::ProposalBondMinimum::get().max(T::ProposalBond::get() * value);
+		T::ProposalBondMaximum::get().map(|m| r = r.min(m));
```

Comment:
Small style question: back in my Mozilla volunteering contributions days, the preferred style to access inner values of monads is via the `if let` or `match` syntax when there is a side effect, should we follow that convention here? It would make the code above look like the following:

```suggestion
		if let Some(m) = T::ProposalBondMaximum::get() {
			r = r.min(m);
		}
```